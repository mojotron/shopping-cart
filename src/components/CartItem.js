import "../styles/CartItem.css";

const CartItem = (props) => {
  return (
    <div className="CartItem">
      <img className="CartItem__icon" src={props.data.image} alt="" />
      <div className="CartItem__info">
        <h3>{props.data.name}</h3>
        <p>{props.data.type}</p>

        <div className="CartItem__quantity">
          <button
            onClick={props.handleDecrementItemQuantity}
            data-id={props.data.id}
          >
            -
          </button>
          <p>{props.data.quantity}</p>
          <button
            onClick={props.handleIncrementItemQuantity}
            data-id={props.data.id}
          >
            +
          </button>
        </div>
      </div>

      <div className="CartItem__price">
        <p>${props.data.price}</p>
      </div>
    </div>
  );
};

export default CartItem;
