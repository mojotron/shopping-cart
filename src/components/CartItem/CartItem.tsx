import React from 'react';
import { CartItemType } from '../../types/cartItemType';
import ItemQuantityControl from '../ItemQuantityControl/ItemQuantityControl';

const CartItem = ({ item }: { item: CartItemType }) => {
  const totalPrice = item.price * item.quantity;
  return (
    <li className="flex gap-4 bg-emerald-50 p-1 rounded-md">
      <img
        className="w-24 h-full object-cover"
        src={item.imageUrl}
        alt={item.name}
      />
      <div className="flex flex-col mt-4">
        <h3 className="capitalize text-xl font-righteous text-emerald-600">
          {item.name}
        </h3>
        <div className="text-sm text-neutral-500 flex flex-col mt-2">
          <p>quantity {item.quantity}</p>
          <p>selling unit 1 {item.sellingUnit}</p>
          <p>price per unit {item.price.toFixed(2)} €</p>
        </div>
      </div>
      <div className="ml-auto flex flex-col justify-center items-center gap-2 font-bold text-neutral-600">
        <p className="text-xl">{totalPrice.toFixed(2)} €</p>
        <ItemQuantityControl item={item} />
      </div>
    </li>
  );
};

export default CartItem;
