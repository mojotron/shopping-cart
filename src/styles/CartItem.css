.CartItem {
  border: 1px solid black;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 50rem;
  /*TEMP*/
  font-size: 2rem;
}
.CartItem__icon {
  height: 15rem;
}
.CartItem__info {
}
.CartItem__quantity {
  display: flex;
  align-items: center;
}
