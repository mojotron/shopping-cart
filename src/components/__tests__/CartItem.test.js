import { render, screen } from "@testing-library/react";
import CartItem from "../CartItem";
import "@testing-library/jest-dom";
import propsMock from "../../mocks/cartItemPropsData.json";

describe("CartItem component", () => {
  test("render CartItem snapshot with mock", () => {
    const { container } = render(<CartItem data={propsMock} />);
    expect(container).toMatchSnapshot();
  });

  test("renders item image", () => {
    render(<CartItem data={propsMock} />);
    expect(screen.getByRole("img").src).toBe("http://localhost/image.jpg");
  });

  test("renders item name", () => {
    render(<CartItem data={propsMock} />);
    expect(screen.getByRole("heading").textContent).toBe("temp");
  });

  test("renders item price", () => {
    render(<CartItem data={propsMock} />);
    expect(screen.getByTitle("item-type").textContent).toBe("backpack");
  });

  test("renders item quantity", () => {
    render(<CartItem data={propsMock} />);
    expect(screen.getByTitle("item-quantity").textContent).toBe("2");
  });
});
