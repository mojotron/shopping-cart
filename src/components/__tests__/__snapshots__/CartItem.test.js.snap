// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component render CartItem snapshot with mock 1`] = `
<div>
  <div
    class="CartItem"
  >
    <img
      alt="temp"
      class="CartItem__icon"
      src="./image.jpg"
    />
    <div
      class="CartItem__info"
    >
      <h3>
        temp
      </h3>
      <p
        title="item-type"
      >
        backpack
      </p>
      <div
        class="CartItem__quantity"
      >
        <button
          class="btn btn--update-quantity"
          data-id="1"
        >
          -
        </button>
        <p
          title="item-quantity"
        >
          2
        </p>
        <button
          class="btn btn--update-quantity"
          data-id="1"
        >
          +
        </button>
      </div>
    </div>
    <div
      class="CartItem__price"
    >
      <p>
        $
        10
      </p>
    </div>
  </div>
</div>
`;
